src/libsystemd/sd-daemon/sd-daemon.c

```diff
@@ -402,7 +402,7 @@ _public_ int sd_is_mq(int fd, const char *path) {
         struct mq_attr attr;
 
         /* Check that the fd is valid */
-        assert_return(fcntl(fd, F_GETFD) >= 0, -errno);
+        assert_return(fd_validate(fd) >= 0, -errno);
```

Comment:
`fd_validate()` returns `-EBADF` if `fd < 0` without touching `errno`. Doesn't this return an uninitialized `errno` in this case?